src/index.ts

```diff
@@ -80,13 +80,22 @@ export const traceProvider: TraceProvider = {
   },
 
   /**
-   * @description Used to propagate traces. Accepts a serialised trace
-   * metadata string and returns a traceContext instance that can be passed
-   * to the wrapHandler.
+   * @description Used to propagate traces. Accepts serialised trace
+   * metadata and returns a traceContext instance that can be passed
+   * to wrapHandler.
    */
   deserializeTraceMetadata: async (
-    traceMetadata: string
-  ): Promise<unknown> => ({
-    distributedTraceHeaders: JSON.parse(traceMetadata),
-  }),
+    traceMetadata: unknown
+  ): Promise<unknown> => {
+    try {
+      if (typeof traceMetadata !== "string") {
```

Comment:
What if this is an object? that is an already deserialised trace meta data?

Do we want to log the invalid data so that we can track down why we're getting it?

Reply:
Yes. I think this needs to take a logger instance (like steveo does).